docs/design/datamover.md

```diff
@@ -0,0 +1,227 @@
+---
+title: datamover_crd_design
+authors:
+  - "@savitharaghunathan"
+reviewers:
+  - "@dymurray"
+  - "@eemcmullan"
+  - "@shubham-pampattiwar"
+approvers:
+  - "@dymurray"
+  - "@eemcmullan"
+  - "@shubham-pampattiwar"
+creation-date: 2022-05-03
+status: provisional
+---
+
+# Data Mover CRD design
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+
+## Summary
+Velero currently supports backup and restore of applications backed by CSI volumes by leveraging the CSI plugin. The problem with CSI snapshots on some providers is that these snapshots are local to the K8s cluster and cannot be recovered if the cluster gets deleted accidentally or if there is a disaster. In order to overcome this issue, a Data Mover solution is made available for users to save the snapshots in a remote storage. 
+
+## Motivation
+
+Create an extensible design to support various data movers that can be integrated with the extended Velero CSI plugin. Vendors should be able to bring their own data mover controller and implementation, and use that with Velero CSI plugin.
+
+## Goals
+* Create an extensible data mover solution
+* Supply a default data mover option 
+* Supply APIs for DataMover CRs (eg: DataMoverBackup, DataMoverRestore, DataMoverClass)
+* Supply a sample codebase for the Data Mover plugin and controller implementation
+
+
+## Non Goals
+* Maintain 3rd party data mover implementations
+* Adding a status watch controller to Velero
+
+## User stories
+
+Story 1: 
+As an application developer, I would like to save the CSI snaphots in a S3 bucket. 
+
+Story 2:
+As a cluster admin, I would like to be able to restore CSI snapshots if disaster happens.
+
+## Design & Implementation details
+
+This design supports adding the data mover feature to the Velero CSI plugin and facilitates integrating various vendor implemented data movers. 
+
+![DataMover CRD](../images/datamovercrd.png)
+
+Note: We will be supporting VolSync as the default data mover. 
+
+There will be two controllers - DataMoverBackup & DataMoverRestore. The DataMoverBackup Controller will be responsible for reconciling DataMoverBackup CR. Likewise, DataMoverRestore Controller will watch for DataMoverRestore CR. Both of these CRs will have a reference to a DataMoverClass. 
```

Comment:
NIT: `DataMoverBackup` -> `VolumeSnapshotBackup` and `DataMoverRestore` -> `VolumeSnapshotRestore` for all the occurrences